boa/vyper/contract.py

```diff
@@ -387,15 +369,35 @@ def __init__(
 
         self.env.register_contract(self.address, self)
 
-        fns = {fn.name: fn for fn in self.global_ctx._function_defs}
-        # add all functions from the interface to the contract
-
+        # add all exposed functions from the interface to the contract
+        # exclude internal functions since
+        # 1. they cannot be called with self._internal_method since internal
+        #    methods are not exposed in the runtime bytecode (but are included
+        #    in the deployed bytecode).
+        # 2. [TBD] Internal methods are namespaced into self.internal_methods.
+        external_fns = {
+            fn.name: fn for fn in self.global_ctx._function_defs if
+            not fn._metadata["type"].is_internal
+        }
+
+        # set external methods as class attributes:
         self._ctor = None
-        if "__init__" in fns:
-            self._ctor = VyperFunction(fns.pop("__init__"), self)
+        if "__init__" in external_fns:
+            self._ctor = VyperFunction(external_fns.pop("__init__"), self)
+
+        for fn_name, fn in external_fns.items():
+            setattr(self, fn_name, VyperFunction(fn, self))
 
-        for fn in fns.values():
-            setattr(self, fn.name, VyperFunction(fn, self))
+        # handle internal methods:
+        internal_fns = {
+            fn.name: fn for fn in self.global_ctx._function_defs if
+            fn._metadata["type"].is_internal
+        }
+
+        # set internal methods as class.internal attributes:
+        self.internal = lambda: None
+        for fn_name, fn in internal_fns.items():
```

Comment:
honestly, instead of having the `internal_fns` variable here, could just loop over `self.global_ctx._function_defs` and skip the ones that are not internal.